homeassistant/components/sensor/hive.py

```diff
@@ -0,0 +1,486 @@
+            return self.get_heating_state_sa()
+        elif self.device_type == "Heating_Mode":
+            return self.get_heating_state_sa()
+        elif self.device_type == "Heating_Boost":
+            s_a = {}
+            if self.session.heating.get_boost(self.node_id) == "ON":
+                minsend = self.session.heating.get_boost_time(self.node_id)
+                s_a.update({"Boost ends in":
+                            (str(minsend) + " minutes")})
+            return s_a
+        elif self.device_type == "HotWater_State":
+            return self.get_hotwater_state_sa()
+        elif self.device_type == "HotWater_Mode":
+            return self.get_hotwater_state_sa()
+        elif self.device_type == "HotWater_Boost":
+            s_a = {}
+            if self.session.hotwater.get_boost(self.node_id) == "ON":
+                endsin = self.session.hotwater.get_boost_time(self.node_id)
+                s_a.update({"Boost ends in":
+                            (str(endsin) + " minutes")})
+            return s_a
+        elif self.device_type == "Hive_Device_BatteryLevel":
+            return None
+        elif self.device_type == "Hive_Device_Sensor":
+            return None
+        elif self.device_type == "Hive_Device_Light_Mode":
+            return None
+        elif self.device_type == "Hive_Device_Plug_Mode":
+            return None
+
+    def get_current_temp_sa(self):
+        """Public get current heating temperature state attributes."""
+        s_a = {}
+        temp_current = 0
+        temperature_target = 0
+        temperature_difference = 0
+
+        if self.node_id in self.session.data.minmax:
+            s_a.update({"Today Min / Max":
+                        str(self.session.data.minmax[self.node_id]
+                            ['TodayMin']) + " °C" + " / "
+                        + str(self.session.data.minmax[self.node_id]
+                              ['TodayMax']) + " °C"})
+
+            s_a.update({"Restart Min / Max":
+                        str(self.session.data.minmax[self.node_id]
+                            ['RestartMin']) + " °C" + " / "
+                        + str(self.session.data.minmax[self.node_id]
+                              ['RestartMax']) + " °C"})
+
+        temp_current = self.session.heating.current_temperature(self.node_id)
+        temperature_target = self.session.heating.\
+            get_target_temperature(self.node_id)
+
+        if temperature_target > temp_current:
+            temperature_difference = temperature_target - temp_current
+            temperature_difference = round(temperature_difference, 2)
+
+            s_a.update({"Current Temperature":
+                        temp_current})
+            s_a.update({"Target Temperature":
+                        temperature_target})
+            s_a.update({"Temperature Difference":
+                        temperature_difference})
+
+        return s_a
+
+    def get_heating_state_sa(self):

```

Comment:
Same as above.

Reply:
The data gathering is already in the library. the code here just converts to HA specific formatting